hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -897,28 +897,40 @@ public HoodieCleanMetadata clean(String cleanInstantTime, boolean scheduleInline
     if (!tableServicesEnabled(config)) {
       return null;
     }
-    final Timer.Context timerContext = metrics.getCleanCtx();
-    CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
-        HoodieTimeline.CLEAN_ACTION, () -> rollbackFailedWrites(skipLocking));
-
-    HoodieTable table = createTable(config, hadoopConf);
-    if (config.allowMultipleCleans() || !table.getActiveTimeline().getCleanerTimeline().filterInflightsAndRequested().firstInstant().isPresent()) {
-      LOG.info("Cleaner started");
```

Comment:
Instead of adding lock, should we disable the cleaning service for one of the job ?

Reply:
We cannot ensure that only one job has the cleaning service enabled.

Reply:
I'm worrying about the efficiency. It is not that efficient for concurrent cleaning and not that necessary.

Reply:
Maybe we can add a special clean lock to ensure that only one clean service is executing, skipping the current cleaning when the cleaning service is  executing, this is more efficient.